src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -108,76 +121,41 @@ public static function collectDeprecations($outputFile)
      */
     public function handleError($type, $msg, $file, $line, $context = [])
     {
-        if ((E_USER_DEPRECATED !== $type && E_DEPRECATED !== $type) || self::MODE_DISABLED === $mode = $this->getMode()) {
+        if ((E_USER_DEPRECATED !== $type && E_DEPRECATED !== $type) || !$this->getConfiguration()->isEnabled()) {
             $ErrorHandler = self::$utilPrefix.'ErrorHandler';
 
             return $ErrorHandler::handleError($type, $msg, $file, $line, $context);
         }
 
-        $trace = debug_backtrace();
+        $deprecation = new Deprecation($msg, debug_backtrace(), $file);
```

Comment:
$msg is unserialized in Deprecation constructor but unserialized message is not used by DeprecationErrorHandler. Tried to fix it in #31382